### PR TITLE
make promise

### PR DESCRIPTION
**Pull request by Gavrilik**

### CODE REVIEW

File: src/features/user/user.controller.ts [hunk 1]
```diff
@@ -19,30 +19,33 @@ export class UserController {
...
   }
 
   @Get(':id')
-  findOne(@Param('id') id: number) {
+  findOne(@Param('id') id: number): Promise<any> {
     return this.userService.findOne(+id);
   }
 
   @Put(':id')
   @UseGuards(AuthGuard('local'))
```
Comment (dmialeshka) [inline src/features/user/user.controller.ts:38]: change Local to JWT guar

Comment (Gavrilik) [reply src/features/user/user.controller.ts]: change

File: package.json [hunk 2]
```diff
@@ -27,25 +27,32 @@
     "@nestjs/mapped-types": "*",
     "@nestjs/passport": "^8.1.0",
     "@nestjs/platform-express": "^8.0.0",
+    "@nestjs/sequelize": "^8.0.0",
```
Comment (dmialeshka) [inline package.json:30]: do we need it?

File: package.json [hunk 3]
```diff
@@ -27,25 +27,32 @@
...
     "http-status": "^1.5.0",
     "passport": "^0.5.2",
     "passport-jwt": "^4.0.0",
     "passport-local": "^1.0.0",
+    "pg": "^8.7.3",
     "reflect-metadata": "^0.1.13",
     "rimraf": "^3.0.2",
-    "rxjs": "^7.2.0"
+    "rxjs": "^7.2.0",
+    "sequelize": "^6.16.1",
```
Comment (dmialeshka) [inline package.json:42]: the same question

File: src/app.module.ts [hunk 4]
```diff
@@ -1,9 +1,29 @@
...
-  imports: [UserModule, BookModule, AuthModule],
+  imports: [
+    TypeOrmModule.forRoot({
+      type: 'postgres',
+      host: 'localhost',
+      port: 5432,
+      username: 'postgres',
+      password: 'oleg6205854',
+      database: 'postgres',
+      entities: ['src/entity/**/*.ts'],
```
Comment (dmialeshka) [inline src/app.module.ts:18]: are you sure that is correct ?

Comment (dmialeshka) [reply src/app.module.ts:18]: use class import instead of path

File: src/app.module.ts [hunk 5]
```diff
@@ -1,9 +1,29 @@
...
+      autoLoadEntities: true,
+    }),
+    UserModule,
+    BookModule,
+    AuthModule,
+  ],
 })
-export class AppModule {}
+export class AppModule {
+  constructor(private connection: Connection) {}
```
Comment (dmialeshka) [inline src/app.module.ts:28]: do we need it ?

File: src/features/book/book.controller.ts [hunk 6]
```diff
@@ -17,30 +17,33 @@ export class BookController {
   constructor(private readonly bookServise: BookService) {}
 
   @Post()
-  @UseGuards(AuthGuard('local'))
-  create(@Body() createBookDto: CreateBookDto) {
+  @UseGuards(AuthGuard('jwt'))
+  create(@Body() createBookDto: CreateBookDto): Promise<any> {
```
Comment (dmialeshka) [inline src/features/book/book.controller.ts:21]: fix type returns

File: src/features/book/book.controller.ts [hunk 7]
```diff
@@ -17,30 +17,33 @@ export class BookController {
...
     return this.bookServise.create(createBookDto);
   }
 
   @Get()
-  findAll() {
+  findAll(): Promise<any> {
```
Comment (dmialeshka) [inline src/features/book/book.controller.ts:26]: fix type returns

File: src/features/book/book.controller.ts [hunk 8]
```diff
@@ -17,30 +17,33 @@ export class BookController {
...
+  findOne(@Param('id') id: number): Promise<any> {
     return this.bookServise.findOne(+id);
   }
 
   @Put(':id')
-  @UseGuards(AuthGuard('local'))
-  update(@Param('id') id: string, @Body() updateBookDto: UpdateBookDto): void {
+  @UseGuards(AuthGuard('jwt'))
+  update(
+    @Param('id') id: number,
```
Comment (dmialeshka) [inline src/features/book/book.controller.ts:38]: fix id type

File: src/features/book/book.service.ts [hunk 9]
```diff
@@ -1,29 +1,36 @@
 import { Injectable } from '@nestjs/common';
-import { BookJsonRepository } from './book.json-repository';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { User } from '../user/entities/user.entity';
```
Comment (dmialeshka) [inline src/features/book/book.service.ts:4]: do we need it here ?

File: src/features/book/book.service.ts [hunk 10]
```diff
@@ -1,29 +1,36 @@
...
 
-  update(id: number, updateBookDto: UpdateBookDto) {
-    return this.bookJsonRepository.update(+id, updateBookDto);
+  update(id: number, updateBookDto: UpdateBookDto): Promise<Book> {
+    return this.bookRepository.save(updateBookDto);
   }
 
-  remove(id: number) {
-    return this.bookJsonRepository.remove(id);
+  async remove(id: number): Promise<void> {
```
Comment (dmialeshka) [inline src/features/book/book.service.ts:33]: why you use async await here ?

File: src/features/user/user.controller.ts [hunk 11]
```diff
@@ -18,31 +18,34 @@ export class UserController {
...
 
   @Get()
-  findAll() {
+  findAll(): Promise<any> {
     return this.userService.findAll();
   }
 
   @Get(':id')
-  findOne(@Param('id') id: number) {
+  findOne(@Param('id') id: number): Promise<any> {
```
Comment (dmialeshka) [inline src/features/user/user.controller.ts:33]: fix returns type

**Review by dmialeshka — CHANGES_REQUESTED** [review]

**Review by dmialeshka — APPROVED** [review]

**Pull request merged**